synfig-core/src/modules/mod_svg/svg_parser.cpp

```diff
@@ -2033,160 +2000,171 @@ Svg_parser::new_guid(){
-	COLOR_NAME("lightskyblue",135, 206, 250)
-	COLOR_NAME("lightslategray",119, 136, 153)
-	COLOR_NAME("lightslategrey",119, 136, 153)
-	COLOR_NAME("lightsteelblue",176, 196, 222)
-	COLOR_NAME("lightyellow",255, 255, 224)
-	COLOR_NAME("lime", 0, 255, 0)
-	COLOR_NAME("limegreen", 50, 205, 50)
-	COLOR_NAME("linen",250, 240, 230)
-	COLOR_NAME("magenta",255, 0, 255)
-	COLOR_NAME("maroon",128, 0, 0)
-	COLOR_NAME("mediumaquamarine",102, 205, 170)
-	COLOR_NAME("mediumblue", 0, 0, 205)
-	COLOR_NAME("mediumorchid",186, 85, 211)
-	COLOR_NAME("mediumpurple",147, 112, 219)
-	COLOR_NAME("mediumseagreen", 60, 179, 113)
-	COLOR_NAME("mediumslateblue",123, 104, 238)
-	COLOR_NAME("mediumspringgreen", 0, 250, 154)
-	COLOR_NAME("mediumturquoise", 72, 209, 204)
-	COLOR_NAME("mediumvioletred",199, 21, 133)
-	COLOR_NAME("midnightblue", 25, 25, 112)
-	COLOR_NAME("mintcream",245, 255, 250)
-	COLOR_NAME("mistyrose",255, 228, 225)
-	COLOR_NAME("moccasin",255, 228, 181)
-	COLOR_NAME("navajowhite",255, 222, 173)
-	COLOR_NAME("navy", 0, 0, 128)
-	COLOR_NAME("oldlace",253, 245, 230)
-	COLOR_NAME("olive",128, 128, 0)
-	COLOR_NAME("olivedrab",107, 142, 35)
-	COLOR_NAME("orange",255, 165, 0)
-	COLOR_NAME("orangered",255, 69, 0)
-	COLOR_NAME("orchid",218, 112, 214)
-	COLOR_NAME("palegoldenrod",238, 232, 170)
-	COLOR_NAME("palegreen",152, 251, 152)
-	COLOR_NAME("paleturquoise",175, 238, 238)
-	COLOR_NAME("palevioletred",219, 112, 147)
-	COLOR_NAME("papayawhip",255, 239, 213)
-	COLOR_NAME("peachpuff",255, 218, 185)
-	COLOR_NAME("peru",205, 133, 63)
-	COLOR_NAME("pink",255, 192, 203)
-	COLOR_NAME("plum",221, 160, 221)
-	COLOR_NAME("powderblue",176, 224, 230)
-	COLOR_NAME("purple",128, 0, 128)
-	COLOR_NAME("red",255, 0, 0)
-	COLOR_NAME("rosybrown",188, 143, 143)
-	COLOR_NAME("royalblue", 65, 105, 225)
-	COLOR_NAME("saddlebrown",139, 69, 19)
-	COLOR_NAME("salmon",250, 128, 114)
-	COLOR_NAME("sandybrown",244, 164, 96)
-	COLOR_NAME("seagreen", 46, 139, 87)
-	COLOR_NAME("seashell",255, 245, 238)
-	COLOR_NAME("sienna",160, 82, 45)
-	COLOR_NAME("silver",192, 192, 192)
-	COLOR_NAME("skyblue",135, 206, 235)
-	COLOR_NAME("slateblue",106, 90, 205)
-	COLOR_NAME("slategray",112, 128, 144)
-	COLOR_NAME("slategrey",112, 128, 144)
-	COLOR_NAME("snow",255, 250, 250)
-	COLOR_NAME("springgreen", 0, 255, 127)
-	COLOR_NAME("steelblue", 70, 130, 180)
-	COLOR_NAME("tan",210, 180, 140)
-	COLOR_NAME("teal", 0, 128, 128)
-	COLOR_NAME("thistle",216, 191, 216)
-	COLOR_NAME("tomato",255, 99, 71)
-	COLOR_NAME("turquoise", 64, 224, 208)
-	COLOR_NAME("violet",238, 130, 238)
-	COLOR_NAME("wheat",245, 222, 179)
-	COLOR_NAME("white",255, 255, 255)
-	COLOR_NAME("whitesmoke",245, 245, 245)
-	COLOR_NAME("yellow",255, 255, 0)
-	COLOR_NAME("yellowgreen",154, 205, 50)
-	return 0;
+Svg_parser::getColor(String name, int position) {
+	if (position < 1 || position > 3) return 0;
+
+	struct RGB {
+		std::uint8_t color[3];
+	};
+
+	static std::unordered_map<std::string, RGB> color_names = {
```

Comment:
Why unordered?

Reply:
In this case, it's just faster. Benchmark:
**unordered_map**
```
Elapsed time: 383 ms
Elapsed time: 382 ms
Elapsed time: 384 ms
```

**map**
```
Elapsed time: 588 ms
Elapsed time: 586 ms
Elapsed time: 583 ms
```

**benchmark.cpp (compiled with -O3)**
```
    const std::string colors[] = {"yellow", "springgreen", "white", "tomato", "mediumspringgreen"};
    int colors_count = sizeof(colors) / sizeof(colors[0]);

    std::srand(std::time(nullptr)); // use current time as seed for random generator

    milliseconds ms = duration_cast< milliseconds >(system_clock::now().time_since_epoch());

    long result = 0;

    for (int i=0; i < 1000*1000*10; ++i) {
      result = getColor(colors[rand() % colors_count], 1);
    }

    milliseconds end = duration_cast< milliseconds >(system_clock::now().time_since_epoch());

    std::cout << "Elapsed time: " << (end - ms).count() << " ms" << std::endl;
```

Reply:
Oh, I wondered sorted `map` would be faster to lookup :O (slower on inserting data though)